dapp/src/hooks/useReferral.ts

```diff
@@ -14,39 +12,19 @@ type UseReferralReturn = {
 
 export default function useReferral(): UseReferralReturn {
   const [referral, setReferral] = useLocalStorage<number>(
-    "referral",
+    "acre.referral",
     env.REFERRAL,
   )
-  const { openModal } = useModal()
-  const { isEmbed } = useIsEmbed()
+  const { isEmbed, embeddedApp } = useIsEmbed()
 
   const detectReferral = useCallback(() => {
-    const param = referralProgram.getReferralFromURL()
-
-    if (isEmbed) {
-      // TODO: Set correct referral for embedded dApp
-      setReferral(0)
-      return
-    }
-
-    if (param === null) {
-      setReferral(env.REFERRAL)
+    if (isEmbed && embeddedApp) {
+      setReferral(referralProgram.getReferralByEmbeddedApp(embeddedApp))
       return
     }
 
-    const convertedReferral = Number(param)
-
-    if (referralProgram.isValidReferral(convertedReferral)) {
-      setReferral(convertedReferral)
-    } else {
-      console.error("Incorrect referral")
-      setReferral(null)
-      openModal(MODAL_TYPES.UNEXPECTED_ERROR, {
-        withCloseButton: false,
-        closeOnEsc: false,
-      })
-    }
```

Comment:
If you remove this part, the referral for the standalone dApp is not set correctly. Are we sure we want to disable the referral program for the standalone page?